apps/prairielearn/assets/scripts/question.ts

```diff
@@ -123,13 +123,66 @@ function fetchResults(socket: Socket, submissionId: string | number) {
     function (msg: any) {
       // We're done with the socket for this incarnation of the page
       socket.close();
+      if (msg.extraHeadersHtml) {
+        const parser = new DOMParser();
+        const headers = parser.parseFromString(msg.extraHeadersHtml, 'text/html');
+
+        const newImportMap = headers.querySelector<HTMLScriptElement>('script[type="importmap"]');
+        if (newImportMap != null) {
+          const currentImportMap = document.head.querySelector<HTMLScriptElement>(
+            'script[type="importmap"]',
+          );
+          if (!currentImportMap) {
+            document.head.appendChild(newImportMap);
+          } else {
+            // This case is not currently possible with existing importmap
+            // functionality. Once an existing importmap has been created, the
+            // importmap cannot be modified. Once this functionality exists this
+            // code can be modified to update the importmap.
+            // https://html.spec.whatwg.org/multipage/webappapis.html#import-map-processing-model
+            const newImportMapJson = JSON.parse(newImportMap.textContent || '{}');
+            const currentImportMapJson = JSON.parse(currentImportMap.textContent || '{}');
+            const newImportMapKeys = Object.keys(newImportMapJson.imports || {}).filter(
+              (key) => !(key in currentImportMapJson.imports),
+            );
+            if (newImportMapKeys.length > 0) {
+              console.warn(
+                'Cannot update importmap. New importmap has imports not in current importmap: ',
+                newImportMapKeys,
+              );
+            }
```

Comment:
At this point dynamic dependencies are not used often enough for this to be caused very often. That said, would it make sense to either cause a refresh or offer one with a modal? I don't think it's needed at this point, but I'm open to discuss.

Reply:
Discussion moved to #9639.